libs/execution_context/unit/execution_context_unit_test.cpp

```diff
@@ -1937,6 +1937,139 @@ TEST_CASE("EBPF_OPERATION_LOAD_NATIVE_MODULE short header", "[execution_context]
+    const cxplat_utf8_string_t section_name{(uint8_t*)("bar"), 3};
+    const ebpf_program_parameters_t program_parameters{
+        EBPF_PROGRAM_TYPE_TEST, EBPF_ATTACH_TYPE_SAMPLE, program_name, section_name};
+    program_ptr program;
+    {
+        ebpf_program_t* local_program = nullptr;
+        REQUIRE(
+            ebpf_program_create(&program_parameters, &local_program) ==
+            (expected_to_succeed ? EBPF_SUCCESS : EBPF_EXTENSION_FAILED_TO_LOAD));
+        program.reset(local_program);
+        if (expected_to_succeed) {
+            helper_function_address_t addresses[1] = {};
+            uint32_t helper_function_ids[] = {EBPF_MAX_GENERAL_HELPER_FUNCTION + 1};
+            REQUIRE(
+                ebpf_program_set_helper_function_ids(
+                    program.get(), EBPF_COUNT_OF(helper_function_ids), helper_function_ids) == EBPF_SUCCESS);
+            REQUIRE(
+                ebpf_program_get_helper_function_addresses(
+                    program.get(), EBPF_COUNT_OF(helper_function_ids), addresses) == EBPF_SUCCESS);
+            REQUIRE(addresses[0].address != 0);
+        }
+    }
+
+    NmrDeregisterProvider(nmr_provider_handle);
+}
+
+TEST_CASE("INVALID_PROGRAM_DATA", "[execution_context][negative]")
+{
+    _ebpf_core_initializer core;
+    core.initialize();
+
+    auto provider_attach_client_callback =
+        [](HANDLE, void*, const NPI_REGISTRATION_INSTANCE*, void*, const void*, void**, const void**) -> NTSTATUS {
+        return STATUS_SUCCESS;
+    };
+    auto provider_detach_client_callback = [](void*) -> NTSTATUS { return STATUS_SUCCESS; };
+
+    NPI_MODULEID module_id = {sizeof(NPI_MODULEID), MIT_GUID, EBPF_PROGRAM_TYPE_TEST_GUID};
+
+    NPI_PROVIDER_CHARACTERISTICS provider_characteristics{
+        0,
+        sizeof(NPI_PROVIDER_CHARACTERISTICS),
+        provider_attach_client_callback,
+        provider_detach_client_callback,
+        nullptr,
+        {
+            0,
+            sizeof(NPI_REGISTRATION_INSTANCE),
+            &EBPF_PROGRAM_INFO_EXTENSION_IID,
+            &module_id,
+            0,
+            nullptr,
+        },
+    };
+
+    // Program data in the provider characterstics struct is NULL.
+    test_register_provider(&provider_characteristics);
+
+    // Add an empty program data.
+    ebpf_program_data_t _test_program_data = {EBPF_PROGRAM_DATA_HEADER, nullptr, nullptr, nullptr, nullptr, nullptr};
+    provider_characteristics.ProviderRegistrationInstance.NpiSpecificCharacteristics = &_test_program_data;
+    test_register_provider(&provider_characteristics);
+
+    // Add an invalid helper function addresses struct to the program data.
+    ebpf_helper_function_addresses_t helper_function_addresses = {EBPF_HELPER_FUNCTION_ADDRESSES_HEADER, 1, nullptr};
+    _test_program_data.program_type_specific_helper_function_addresses = &helper_function_addresses;
+    test_register_provider(&provider_characteristics);
```

Comment:
nit: should we make every other field (except `program_type_specific_helper_function_addresses`) in `_test_program_data` to a valid value?

same for other negative scenarios tested in this function.

Reply:
Fixed.